internal/stores/s3_test.go

```diff
@@ -229,3 +229,77 @@ func TestS3StoreRetrieve(t *testing.T) {
 	assert.Equal(t, `tla`, string(b))
 
 }
+
+func TestS3GetBucketNameWithS3Prefix(t *testing.T) {
```

Comment:
asking to learn: why do we need this test? we are already testing backendAddresses with s3 prefix: https://github.com/discentem/cavorite/blob/1b7c668ab587ab2dd260c2cd051d71c0cfb6316d/internal/stores/s3_test.go#L165